vault/resource_database_secret_backend_connection.go

```diff
@@ -23,66 +24,124 @@ type connectionStringConfig struct {
-		dbBackendMSSQL:         {name: dbBackendMSSQL},
-		dbBackendMongoDB:       {name: dbBackendMongoDB},
-		dbBackendMongoDBAtlas:  {name: dbBackendMongoDBAtlas},
-		dbBackendMySQL:         {name: dbBackendMySQL},
-		dbBackendMySQLAurora:   {name: dbBackendMySQLAurora},
-		dbBackendMySQLLegacy:   {name: dbBackendMySQLLegacy},
-		dbBackendMySQLRDS:      {name: dbBackendMySQLRDS},
-		dbBackendPostgres:      {name: dbBackendPostgres},
-		dbBackendOracle:        {name: dbBackendOracle},
-		dbBackendSnowflake:     {name: dbBackendSnowflake},
-		dbBackendRedshift:      {name: dbBackendRedshift},
+	dbEngineCassandra = &dbEngine{
+		name:              "cassandra",
+		defaultPluginName: "cassandra" + dbPluginSuffix,
+	}
+	dbEngineCouchbase = &dbEngine{
+		name:              "couchbase",
+		defaultPluginName: "couchbase" + dbPluginSuffix,
+	}
+
+	dbEngineElasticSearch = &dbEngine{
+		name:              "elasticsearch",
+		defaultPluginName: "elasticsearch" + dbPluginSuffix,
+	}
+	dbEngineHana = &dbEngine{
+		name:              "hana",
+		defaultPluginName: "hana" + dbPluginSuffix,
+	}
+	dbEngineInfluxDB = &dbEngine{
+		name:              "influxdb",
+		defaultPluginName: "influxdb" + dbPluginSuffix,
+	}
+	dbEngineMSSQL = &dbEngine{
+		name:              "mssql",
+		defaultPluginName: "mssql" + dbPluginSuffix,
+	}
+	dbEngineMongoDB = &dbEngine{
+		name:              "mongodb",
+		defaultPluginName: "mongodb" + dbPluginSuffix,
+	}
+	dbEngineMongoDBAtlas = &dbEngine{
+		name:              "mongodbatlas",
+		defaultPluginName: "mongodbatlas" + dbPluginSuffix,
+	}
+	dbEngineMySQL = &dbEngine{
+		name:              "mysql",
+		defaultPluginName: "mysql" + dbPluginSuffix,
+	}
+	dbEngineMySQLAurora = &dbEngine{
+		name:              "mysql_aurora",
+		defaultPluginName: "mysql-aurora" + dbPluginSuffix,
+	}
+	dbEngineMySQLLegacy = &dbEngine{
+		name:              "mysql_legacy",
+		defaultPluginName: "mysql-legacy" + dbPluginSuffix,
+	}
+	dbEngineMySQLRDS = &dbEngine{
+		name:              "mysql_rds",
+		defaultPluginName: "mysql-rds" + dbPluginSuffix,
+	}
+	dbEnginePostgres = &dbEngine{
+		name:              "postgresql",
+		defaultPluginName: "postgresql" + dbPluginSuffix,
+	}
+	dbEngineOracle = &dbEngine{
+		name:              "oracle",
+		defaultPluginName: "oracle" + dbPluginSuffix,
+	}
+	dbEngineSnowflake = &dbEngine{
+		name:              "snowflake",
+		defaultPluginName: "snowflake" + dbPluginSuffix,
+	}
+	dbEngineRedshift = &dbEngine{
+		name:              "redshift",
+		defaultPluginName: "redshift" + dbPluginSuffix,
+	}
+
+	dbEngines = []*dbEngine{
+		dbEngineCassandra,
+		dbEngineCouchbase,
+		dbEngineElasticSearch,
+		dbEngineHana,
+		dbEngineInfluxDB,
+		dbEngineMSSQL,
+		dbEngineMongoDB,
+		dbEngineMongoDBAtlas,
+		dbEngineMySQL,
+		dbEngineMySQLAurora,
+		dbEngineMySQLLegacy,
+		dbEngineMySQLRDS,
+		dbEnginePostgres,
+		dbEngineOracle,
+		dbEngineSnowflake,
+		dbEngineRedshift,
 	}
 )
 
 type dbEngine struct {
```

Comment:
Small nit: It shouldn't really matter too much since this referenced within the same package (thus fields will technically still be accessible), but since these fields are never meant to be changed, I wonder if we should expose getters, i.e. `Name()` and `DefaultPluginName()`, so that we don't end up accidentally overriding them down the road on future references.

Reply:
Updated in https://github.com/hashicorp/terraform-provider-vault/pull/1326/commits/aac2869670f393b557d9c949363b8a7e10a5fd7b